core/include/vecmem/edm/details/accessor_traits.hpp

```diff
@@ -0,0 +1,280 @@
+/* VecMem project, part of the ACTS project (R&D line)
+ *
+ * (c) 2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+#pragma once
+
+// Local include(s).
+#include "vecmem/edm/details/device_traits.hpp"
+#include "vecmem/edm/details/host_traits.hpp"
+#include "vecmem/edm/schema.hpp"
+
+// System include(s).
+#include <tuple>
+
+namespace vecmem::edm::details {
+
+/// @name Traits for figuring out the return value for a given variable type
+/// @{
+
+template <typename TYPE>
+struct accessor_host_type {
+    struct UNKNOWN_TYPE {};
+    using stored_type = UNKNOWN_TYPE;
+    using return_type = UNKNOWN_TYPE;
+    using const_return_type = UNKNOWN_TYPE;
```

Comment:
Also, is there a good reason to add these `UNKNOWN_TYPE` typedefs here rather than the more canonical approach of just not defining these types at all?

Reply:
Very true. I'm very open to changing this. I was just trying to find a formalism that may produce the most readable error messages. :thinking:

Reply:
Well the most readable would certainly be to use concepts. :smile: 

I suspect that it doesn't really matter because these are type traits for internal use (right?) but not defining the types gives a bit more flexibility when using SFINAE (e.g. the type not existing can trigger a specialization failure, but the unknown type can't).